src/site-apps.ts

```diff
@@ -74,18 +72,33 @@ export class SiteApps {
         return globals
     }
 
-    afterDecideResponse(response?: DecideResponse): void {
-        if (isArray(response?.siteApps) && response.siteApps.length > 0) {
-            if (this.enabled && this.instance.config.opt_in_site_apps) {
+    loadIfEnabled() {
+        if (
+            this._decideServerResponse &&
+            isArray(this._decideServerResponse) &&
+            this._decideServerResponse.length > 0
+        ) {
+            // can't use if site apps are disabled, or if we're not asking /decide for site apps
+            const enabled = this.instance.config.opt_in_site_apps && !this.instance.config.advanced_disable_decide
+            if (enabled) {
                 const checkIfAllLoaded = () => {
                     // Stop collecting events once all site apps are loaded
                     if (this.appsLoading.size === 0) {
                         this.loaded = true
                         this.missedInvocations = []
                     }
                 }
-                for (const { id, url } of response['siteApps']) {
-                    // TODO: if we have opted out and "type" is "site_destination", ignore it... but do include "site_app" types
+                for (const { id, type, url } of this._decideServerResponse) {
+                    // if consent isn't given, skip site destinations
+                    if (this.instance.consent.isOptedOut() && type === 'site_destination') {
+                        checkIfAllLoaded()
+                        continue
+                    }
+                    // if the site app is already loaded, skip it
+                    if (!isUndefined(assignableWindow[`__$$ph_site_app_${id}_posthog`])) {
+                        checkIfAllLoaded()
+                        continue
+                    }
```

Comment:
Nit: I'd just `continue` here, and do a global `checkIfAllLoaded()` below the loop.